engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/CallerFieldOrMethodVisitor.java

```diff
@@ -22,11 +19,10 @@
     CallerFieldOrMethodVisitor(Method m, JavaClass jc, Context c) {
         super(m, jc, c);
         cp = new ConstantPoolGen(jc.getConstantPool());
-        boolean testMethod = getAnnotation(jc.getConstantPool(), m::getAnnotationEntries, JUNIT_PACKAGE, TESTING_PACKAGE) != null;
 
-        boolean ignore = Objects.nonNull(AnnotationUtils.getAnnotation(jc.getConstantPool(), jc::getAnnotationEntries, Annotations.TEST.IGNORE.getDesc()));
-        if (!ignore)
-            ignore = Objects.nonNull(AnnotationUtils.getAnnotation(jc.getConstantPool(), m::getAnnotationEntries, Annotations.TEST.IGNORE.getDesc()));
+        boolean testMethod = getAnnotation(jc.getConstantPool(), m::getAnnotationEntries, ALL_TEST_PACKAGES).isPresent();
+
+        boolean ignore = AnnotationUtils.isIgnore(jc, jc::getAnnotationEntries) || AnnotationUtils.isIgnore(jc, m::getAnnotationEntries);
```

Comment:
as improvement -- **ignore** is only needed for when testMethod is true
it can be evaluated only when 
if (testMethod){ .... }

Reply:
Noted for future improvement!